File: lab 7/scala/src/main/scala/AsymetricPhilosopher.scala
class AsymetricPhilosopher(val number: Int, val f1: Fork, val f2: Fork)
  extends Philosopher(number, f1, f2)
{

  @throws[InterruptedException]
  override protected def eat(): Unit = {
    if (number % 2 == 0) consume(f1, f2)
    else consume(f2, f1)
  }
}

File: lab 7/java/src/AsymetricPhilosopher.java
public class AsymetricPhilosopher extends Philosopher
{
    public AsymetricPhilosopher(int number, Fork f1, Fork f2)
    {
        this.number = number;
        this.f1 = f1;
        this.f2 = f2;
    }

    protected void eat() throws InterruptedException
    {
        if(number % 2 == 0)
        {
            consume(f1, f2);
        }
        else
        {
            consume(f2, f1);
        }
    }
}
